### PR TITLE
E2E tests: Add documentation

### PR DESCRIPTION
**Pull request by juliamagan**
|Related issue|
|-------------|
|     #2983         |

## Description
The tests developed in #2872 have been documented.

### Updated

- Rename `basic_cases` to `test_basic_cases`
- Rename `test_netcat` to `test_detecting_suspicious_binaries`
- Add documentation to every E2E test
- Add necessary tags to qa-docs schema
- Fix some errors in qa-docs config

---

## Testing performed

- `qa-docs -p tests/end_to_end`:

```
root@pop-os:/home/juliamagan/Desktop/QA/wazuh-qa# qa-docs -p tests/end_to_end/
2022-08-18 12:42:35,499 - INFO - Parsing all tests located in tests/end_to_end/
2022-08-18 12:42:36,573 - INFO - Run completed, documentation location: /tmp/qa_docs/output
```
Generated documentation:  [test_basic_cases.zip](https://github.com/wazuh/wazuh-qa/files/9372754/test_basic_cases.zip)

- `qa-docs -p tests -t end_to_end`:

```
root@pop-os:/home/juliamagan/Desktop/QA/wazuh-qa# qa-docs -p tests/ -t end_to_end 
2022-08-18 12:46:44,976 - INFO - Parsing the following test(s) type(s): ['end_to_end']
2022-08-18 12:46:44,991 - INFO - Running QADOCS
2022-08-18 12:46:46,037 - INFO - Run completed, documentation location: /tmp/qa_docs/output
```

Generated documentation: [test_basic_cases.zip](https://github.com/wazuh/wazuh-qa/files/9372761/test_basic_cases.zip)

### CODE REVIEW

**Review by roronoasins — APPROVED** [review]
LGTM

File: tests/end_to_end/README.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# wazuh-qa
+
+Wazuh - Quality assurance automation templates
```
Comment (jmv74211) [inline tests/end_to_end/README.md:3]: copy, paste? I think it is not appropriate. Instead, you could put an introduction to what E2E tests are and, perhaps, the classification we make within this structure.

File: tests/end_to_end/test_basic_cases/test_yara_integration/test_yara_integration.py [hunk 2]
```diff
@@ -32,6 +71,44 @@
 @pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
 def test_yara_integration(configure_environment, metadata, get_dashboard_credentials, generate_events,
                           clean_alerts_index):
+    '''
+    description: Check that an alert is generated when malware is downloaded.
+
+    test_phases:
+        - Set a custom Wazuh configuration.
+        - Download to generate the event.
```
Comment (jmv74211) [inline tests/end_to_end/test_basic_cases/test_yara_integration/test_yara_integration.py:79]: Download what?

Comment (juliamagan) [reply tests/end_to_end/test_basic_cases/test_yara_integration/test_yara_integration.py]: Done in 6c5bafd

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**